subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -461,6 +461,10 @@ tmpdir is currently ${System.getProperty("java.io.tmpdir")}""")
         failure.assertHasCause(cause)
     }
 
+    protected void failureHasCause(Matcher<? super String> matcher) {
+        failure.assertThatCause(matcher)
+    }
```

Comment:
I understand that this is nice and general, but for readability I think accepting `Pattern` specifically here would lead to nicer code in the tests itself as you can use the `/regex/` syntax instead.

Reply:
Removed and used `Pattern`.